examples/all-clusters-app/all-clusters-common/gen/encoder.cpp

```diff
@@ -0,0 +1,18 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+// THIS FILE IS GENERATED BY ZAP
```

Comment:
That would be better I agree. Currently `src/app/zap-templates/app-templates.json` declare a static list of files to be generated, and this particular file is empty because of `{{#if (chip_has_client_clusters)}}` in the template itself.

Fixing that would require to update `src/app/chip_data_model.gni` to generate a specific `templates.json` file based on the content of the `zap` file if there is no client clusters at all. Doable, but definitively out of scope for this particular PR.